src/array/specification.rs

```diff
@@ -75,16 +75,40 @@ pub fn check_offsets_minimal<O: Offset>(offsets: &[O], values_len: usize) -> usi
 /// * any slice of `values` between two consecutive pairs from `offsets` is invalid `utf8`, or
 /// * any offset is larger or equal to `values_len`.
 pub fn check_offsets_and_utf8<O: Offset>(offsets: &[O], values: &[u8]) {
-    offsets.windows(2).for_each(|window| {
-        let start = window[0].to_usize();
-        let end = window[1].to_usize();
-        // assert monotonicity
-        assert!(start <= end);
+    const SIMD_CHUNK_SIZE: usize = 64;
+
+    let all_ascii = values.is_ascii();
+
+    if all_ascii {
+        if offsets.len() == 0 {
+            return;
+        }
+        let mut last = offsets[0];
+        assert!(offsets.iter().skip(1).all(|&end| {
+            let monotone = last <= end;
+            last = end;
+            monotone
+        }));
```

Comment:
we can call (or use this impl instead) `check_offsets(offsets, values.len())` here.